### DISCUSSION TITLE
Notification when download ends

### DISCUSSION THREAD

**Discussion by TonioIncredible** [PHPOffice/PhpSpreadsheet#3248] [Q&A]
Hello,

I download a file using **$writer->save('php://output');** (after specifying the correct header), and it works well.

But I would like to be able to do more treatment when the download is finished, how can I do? I cannot send another header.
For instance, I need to hide a "wait" animated gif at the end, or to go to a specific URL.

I searched quite a lot and did not find.

Thanks for your suggestions

**Comment by MarkBaker** [comment]
Consider how you might do this for any file download? Can you do it if, for example, you did something like:
```php
$objInputStream = fopen("/my/textfile.txt", "rb");
$objTempStream = fopen("php://output", "w+b");
stream_copy_to_stream($objInputStream, $objTempStream);
```

This isn't a PhpSpreadsheet-specific issue, not even a PHP-specific issue, but a general question about the `http` protocol, and how that works.


The closest to a solution that I can suggest would be to send an additional `REFRESH` header, that would redirect after the content of the file had been sent.
```php
header('Refresh: 10; URL=http://yoursite.com/thankYouForDownloadingPage.php');
```

But I make no guarantees about how effective this would be

**Comment by TonioIncredible** [comment]
Thank you for your reply.
Yes, I imagine it is not a specific PHPSpreadSheet issue, but I imagine many people should face it.
Your first solution implies that the file is stored on the server, which I don't want.

And the header refresh does not work. I am already sending headers:
		header('Content-Type: application/vnd.openxmlformats-officedocument.spreadsheetml.sheet');
		header("Content-Disposition: attachment;filename=\"$fic\"");
		header('Cache-Control: max-age=0');

**Reply by MarkBaker** [reply]
The first "solution" isn't a solution, it's a question; how would you do it for other file downloads if they weren't generated by PhpSpreadsheet.

It doesn't matter if you're already sending headers; you can add more headers than those required to tell the browser to accept a spreadsheet file download, so you can send a `refresh` header as well. I don't know if sending the `refresh` header will work or not, but try sending it with the Content type and disposition, and the cache control headers, and see what affect it has.

**Reply by TonioIncredible** [reply]
Sorry, Refresh does not work

**Reply by TonioIncredible** [reply]
But thanks for your help
